lib/src/main/java/space/taran/arklib/domain/preview/GeneralPreviewGenerator.kt

```diff
@@ -0,0 +1,50 @@
+package space.taran.arklib.domain.preview
+
+import android.util.Log
+import space.taran.arklib.domain.preview.generator.ImagePreviewGenerator
+import space.taran.arklib.utils.LogTags.PREVIEWS
+import space.taran.arklib.utils.extension
+import space.taran.arklib.utils.getMimeTypeUsingTika
+import space.taran.arklib.domain.preview.generator.PdfPreviewGenerator
+import space.taran.arklib.domain.preview.generator.PreviewGenerator
+import space.taran.arklib.domain.preview.generator.TxtPreviewGenerator
+import java.nio.file.Path
+import kotlin.system.measureTimeMillis
+import space.taran.arklib.domain.preview.generator.VideoPreviewGenerator
+
+object GeneralPreviewGenerator {
+
+    // Use this list to declare new types of generators
+    private val generators = listOf(
```

Comment:
Should we add `LinkPreviewGenerator` here?

Reply:
No, we discussed with @mdrlzy . ```LinkPreviewGenerator``` seems to be broken, so we decided to remove it, I think temporarily